sdk/python/kfp/compiler/compiler_utils_test.py

```diff
@@ -106,7 +106,68 @@ def test_additional_input_name_for_pipeline_channel(self, channel,
                 },
                 'replica_count': 1
             }],
-        },)
+        },
+        {
+            'data': [{
+                'first_name':
+                    f'my_first_name: {str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="first_name"))}',
+                'last_name':
+                    f'my_last_name: {str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="last_name"))}',
+            }],
+            'old_value':
+                '{{channel:task=first_name;name=Output;type=String;}}',
+            'new_value':
+                '{{$.inputs.parameters['
+                'pipelinechannel--first_name-Output'
+                ']}}',
+            'expected': [{
+                'first_name':
+                    'my_first_name: {{$.inputs.parameters[pipelinechannel--first_name-Output]}}',
+                'last_name':
+                    f'my_last_name: {str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="last_name"))}',
+            }],
+        },
```

Comment:
This test case partially replaces the value in the top level. It succeeds by the original implementation.